src/backend/base/langflow/services/utils.py

```diff
@@ -157,17 +164,92 @@ def initialize_session_service() -> None:
     )
 
 
+async def clean_transactions(settings_service: "SettingsService", session: AsyncSession) -> None:
+    """Clean up old transactions from the database.
+
+    This function deletes transactions that exceed the maximum number to keep (configured in settings).
+    It orders transactions by timestamp descending and removes the oldest ones beyond the limit.
+
+    Args:
+        settings_service: The settings service containing configuration like max_transactions_to_keep
+        session: The database session to use for the deletion
+
+    Returns:
+        None
+    """
+    try:
+        # Delete transactions using bulk delete
+        delete_stmt = (
+            delete(TransactionTable)
+            .where(
+                TransactionTable.id.in_(
+                    select(TransactionTable.id)
+                    .order_by(TransactionTable.timestamp.desc())
+                    .offset(settings_service.settings.max_transactions_to_keep)
+                )
+            )
+            .execution_options(synchronize_session="evaluate")
+        )
+
+        await session.exec(delete_stmt)
+        await session.commit()
+        logger.debug("Successfully cleaned up old transactions")
+    except Exception as exc:  # noqa: BLE001
+        logger.error(f"Error cleaning up transactions: {exc!s}")
```

Comment:
Use `logging.exception` ?

Reply:
logging.exception pollutes the logs too much. If something else failed and per chance this one if the middle of the logs, it would confuse both the user and the dev debugging the case.

Reply:
Then maybe the exception catching should be less broad (see BLE001 rule).
In general it is recommended to only catch "expected" exceptions and reraise or log "unknown" exceptions.

Reply:
Done.